src/Microsoft.Performance.SDK/Processing/TableConfiguration.cs

```diff
@@ -197,16 +197,10 @@ public IEnumerable<ColumnConfiguration> Columns
         /// <exception cref="System.ArgumentNullException">
         ///     <paramref name="column"/> is <c>null</c>.
         /// </exception>
-        /// <exception cref="InvalidEnumArgumentException">
-        ///     <paramref name="columnRole"/> is not a valid member of the
-        ///     <see cref="ColumnRole"/> enumeration.
-        ///     - or -
-        ///     <paramref name="columnRole"/> is equal to <see cref="ColumnRole.CountColumnMetadata"/>.
-        /// </exception>
         /// <exception cref="System.InvalidOperationException">
         ///     <paramref name="column"/> is a metadata column (see <see cref="AllMetadataColumns"/>.
         /// </exception>
-        public void AddColumnRole(ColumnRole columnRole, ColumnConfiguration column)
+        public void AddColumnRole(string columnRole, ColumnConfiguration column)
         {
             Guard.NotNull(column, nameof(column));
```

Comment:
should we add a Guard.NotNullOrWhitespace to columnRole? If we do we need to update the docstring too